lib/rules/react-svg-path-precision.js

```diff
@@ -0,0 +1,60 @@
+const t = require("@babel/types");
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Ensure that SVG paths don't use too many decimal places",
+            category: "react",
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    precision: {
+                        type: "number"
+                    }
+                },
+                additionalProperties: false
+            },
+        ],
+    },
+
+    create(context) {
+        let precision = 2;
+        for (const option of context.options) {
+            if (typeof option === "object") {
+                if (option.hasOwnProperty("precision")) {
+                    precision = option.precision;
+                }
+            }
+        }
+        const pattern = `(\\.${"[0-9]".repeat(precision)})[0-9]+`;
+        const regex = new RegExp(pattern, "g");
+
+        return {
+            
+            JSXAttribute(node) {
+                if (t.isJSXAttribute(node) && t.isJSXIdentifier(node.name, {name: "d"})) {
+                    if (t.isJSXOpeningElement(node.parent) && t.isJSXIdentifier(node.parent.name, {name: "path"})) {
+                        const d = node.value.value;
+
+                        if (regex.test(d)) {
```

Comment:
I think it might be better to change the number regex to be something like: `const regex = /\d+\\.\d+/g;` and then to perform the check you'd need to do:

```
const match = d.match(regex);
if (match) {
    const roundedValue = parseFloat(match).toFixed(precision);

    if (match !== roundedValue) {
		context.report({...
```

Reply:
I continue to construct the pattern from a string so that we can use it both for detecting that a `d` string has some numbers have too many decimals as well as limiting which numbers get updated to only those that need to be.